code/components/citizen-resources-core/include/ResourceEventComponent.h

```diff
@@ -130,6 +130,9 @@ class RESOURCES_CORE_EXPORT ResourceEventManagerComponent : public fwRefCountabl
 	//
 	fwEvent<const std::string&, const std::string&, const std::string&> OnQueueEvent;
 
+	// Arguments: eventName, eventPayloadSize, eventSource
+	fwEvent<const std::string_view&, size_t&, const std::optional<std::string_view>&> OnClientEventTriggered;

```

Comment:
A little odd as the concept of server->client events only exists on the server and is part of ServerResources - I wonder if there'd be a better place for this callout across components, though.

Reply:
[ServerEventComponent](https://github.com/citizenfx/fivem/blob/65bf224097b1107c10f84f5dfc25ee8e4bddc95d/code/components/citizen-server-impl/include/ServerEventComponent.h#L14) might be a better place to put something like this and just change the initialization to `OnServerCreate`